src/util/evented.ts

```
@@ -124,7 +124,11 @@ export class Evented {
         const type = event.type;
 
         if (this.listens(type)) {
-            (event as any).target = this;
+            try {
+                (event as any).target = this;
+            } catch (_) { // eslint-disable-line @typescript-eslint/no-unused-vars
+                // On some type of events, setting target raises a TypeError
+            }
```

I also don't like this but I haven't find any other way to fix the error.
Javascript was complaining with `TypeError: Cannot set property target of #<Event> which has only a getter` while running some tests

---

I would look into fixing those tests instead of the code, although I haven't seen these console errors when I ran any of the tests locally.
Where did you see this error?

---

Those errors were getting swallowed by the `catch` in map.ts. Just catching `Abort` errors is what made them show, although they were already present.
I've pushed a fix that removes this try/catch. We were using browser `Event` instead of `Maplibre `Event` in some tests and that was triggering the error

---

Ah, makes sense, thanks!